crates/wasi/src/preview2/preview2/io.rs

```diff
@@ -195,21 +209,162 @@ impl<T: WasiView> streams::Host for T {
         Ok(bytes_spliced)
         */
 
-        todo!()
+        todo!("stream forward is not implemented")
     }
 
     async fn subscribe_to_input_stream(&mut self, stream: InputStream) -> anyhow::Result<Pollable> {
-        Ok(self
-            .table_mut()
-            .push(Box::new(PollableEntry::Read(stream)))?)
+        // Ensure that table element is an input-stream:
+        let _ = self.table_mut().get_input_stream_mut(stream)?;
```

Comment:
Yes this is a pretty significant complication. Right now we have decided to punt on it by adding a big warning in the source code, land this PR, then see if any of the assumptions involved change as we switch over to resources, and then fix it. It would be a crash vector for the whole runtime, but it would also require some pretty major work, so I want to defer it to a future PR. We will fix it before we mark the preview2 implementation as production ready.